src/auth.rs

```diff
@@ -105,9 +91,8 @@ impl SessionInfo {
 
         if response.status().is_success() {
             debug!("response content: {:?}", response);
-            let session_info: SessionInfo =
-                response.json().await.expect("Error parsing session_info");
-            Some(session_info)
+            let session: Session = response.json().await.expect("Error parsing session_info");
+            Some(session)
```

Comment:
You could map `Result` into `Option` using `ok()` but in this case wouldn't it be better to make `fetch_session_data` return a result in case of an `Option`? Then you would propagate the `Error parsing session_info` (maybe you need to change `session_info` to `session`) upwards an let the caller decide.

Reply:
you are right, and I think makes sense to change this method altogether, and set It's signature to `-> Result<T, Err>` instead `Option`